snap/integrity/integrity_test.go

```diff
@@ -0,0 +1,111 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package integrity_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/snap/integrity"
+	"github.com/snapcore/snapd/snap/squashfs"
+	"github.com/snapcore/snapd/testutil"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type IntegrityTestSuite struct {
+	testutil.BaseTest
+}
+
+var _ = Suite(&IntegrityTestSuite{})
+
+func (s *IntegrityTestSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+}
+
+func (s *IntegrityTestSuite) TearDownTest(c *C) {
+	s.BaseTest.TearDownTest(c)
+}
+
+func (s *IntegrityTestSuite) TestAlign(c *C) {
+	align := integrity.Align
+	blockSize := uint64(integrity.BlockSize)
+
+	ret := align(0)
```

Comment:
(super-nitpick) This could be written as a table driven test (which is something we do often):
```
for _, tc := range []struct{
  input uint64
  expectedOutput uint64
} {
  { 0, 0 },
  { 1, blockSize },
  { blockSize+1, 2*blockSize },
} {
  ret = align(tc.input)
  c.Check(ret, Equals, tc.expectedOutput, Commentf("%v", tc))
}
```